File: 1 Python/Codeforces-1834A-UnitArray.py
#using mathematical intuition
t = int(input())

for _ in range(t):
    n = int(input())
    l = list(map(int, input().rstrip().split()))
    lp1 = l.count(1)
    lm1 = l.count(-1)
    operations = 0
    # sum is dependent on the relative counts: relative lp1>=lm1
    # product is dependent upon the even/odd count of -1s
    # we shall strive to first correct the sum 
    # the if conditions should be partially independent

    if(lm1>lp1):
        if (lm1-lp1)%2!=0:
            operations += (lm1+1-lp1)//2 # sum is corrected at each swap by 2 (1-(-1))
        else:
            operations += (lm1-lp1)//2 # sum is corrected at each swap by 2 (1-(-1))
        
    if (lm1-operations)%2!=0: # mindful of post-operations count of -1s
        operations+=1 # for correcting the products
    
    print(operations)

# using list traversal
t = int(input())

for _ in range(t):
    n = int(input())
    l = list(map(int, input().rstrip().split()))
    l.sort()
    operations = 0
    for i in range(n):
        if sum(l)<0 or l.count(-1)%2!=0:
            l[i] = 1
            operations+=1
        else:
            break
    print(operations)

File: 2 CPP/Codeforces-1834A-UnitArray.cpp
//using mathematical intuition
//

#include <bits/stdc++.h>
using namespace std;

void solve(){
    int t;
    cin >> t;
    while(t--){
        int n;
        cin >> n;
        // instead of storing numbers, we shall take just the counts for processing
        int p1 = 0, m1=0;
        for(int i=0;i<n;++i){
            int num;
            cin >> num;
            if (num==1){
                p1+=1;
            }
            else{
                m1+=1;
            }
        }
        
        int operations = 0;
        
        //for sum correction
        if(p1<m1){
            operations = ((m1-p1)%2!=0) ? (m1+1-p1)/2 : (m1-p1)/2;
        }
        operations += (m1-operations)%2; //for product correction
        
        cout << operations << "\n";
    }
}

int main(){
    solve();
    return 0;
}


//using list traversal..
#include <bits/stdc++.h>
using namespace std;

void solve(){
    int t;
    cin >> t;
    while(t--){
        int n;
        cin >> n;
        vector <int> nums;
        for(int i=0;i<n;++i){
            int num;
            cin >> num;
            nums.push_back(num);
        }
        sort(nums.begin(),nums.end());
        int operations = 0;
        for(int j=0;j<n;++j){
            if(accumulate(nums.begin(),nums.end(),0)<0 or count(nums.begin(),nums.end(),-1)%2!=0){
                nums[j]=1;
                operations+=1;
            }
            else{
                break;
            }
        }
        cout << operations << "\n";
    }
}

int main(){
    solve();
    return 0;
}
